benchmarks/profiler/utils/planner_utils.py

```diff
@@ -0,0 +1,162 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+
+from dynamo.planner.utils.planner_argparse import create_sla_planner_parser
+
+
+def add_planner_arguments_to_parser(
+    parser: argparse.ArgumentParser, prefix: str = "planner-"
+):
+    """
+    Dynamically add planner arguments from create_sla_planner_parser() to the given parser.
+    Only adds arguments that don't already exist in the parser (without prefix).
+
+    Args:
+        parser: The ArgumentParser to add arguments to
+        prefix: Prefix to add to planner argument names to avoid conflicts
+    """
+    # Create a temporary planner parser to extract its arguments
+    planner_parser = create_sla_planner_parser()
+
+    # Get existing argument names in the parser (without dashes)
+    existing_dests = {action.dest for action in parser._actions}
+
+    # Add a group for planner arguments
+    planner_group = parser.add_argument_group(
+        "planner arguments",
+        "Arguments that will be passed to the planner service (only showing args not already in profile_sla)",
+    )
+
+    # Iterate through planner parser actions and add them with prefix
+    for action in planner_parser._actions:
+        # Skip help and positional arguments
+        if action.dest in ["help"] or not action.option_strings:
+            continue
+
+        # Skip if this argument already exists in the main parser (without prefix)
+        if action.dest in existing_dests:
+            continue
+
+        # Create new option strings with prefix
+        new_option_strings = [
+            f"--{prefix}{opt.lstrip('-')}" for opt in action.option_strings
+        ]
+
+        # Determine the action type from the class name
```

Comment:
There's a lot of decomposition you could do here to clean up the functions -- like one for store_true/store_false work

Reply:
As an aside, are there existing libraries that can handle this work for you?

Reply:
Can you decompose all of your functions into more helper functions?